### PR TITLE
Fix various strict null errors (part 3)

### PR DESCRIPTION
**Pull request by jpveooys**
## Related issue

#2755

## Overview

This fixes a subset of type errors on master that were being suppressed by `"strictNullChecks": false`. 

## Link to preview

https://5e25c277526d380020b5e418-lylsvjcbge.chromatic.com/

## Reason

> Currently, the TypeScript configuation in `packages/react-component-library/tsconfig.json` contains:
> 
> ```
>     "strictNullChecks": false,
> ```
> 
> This means that types implicitly allow null and undefined. This is undesirable as it means:
> 
> - types are misleading
> - there may be hidden bugs (either due to values being null or undefined unintentionally, or due to null or undefined not being handled)
> - there may be missed test cases

## Work carried out

- [x] Resolve strict null errors for various components

## Developer notes

These should be safe to fix on master (as opposed to `release/3.0.0`).

### CODE REVIEW

File: packages/react-component-library/src/components/RangeSliderE/RangeSliderE.test.tsx [hunk 1]
```diff
@@ -23,8 +23,8 @@ describe('RangeSliderE', () => {
     values: ReadonlyArray<number>,
     data: { activeHandleID: string }
   ) => void
-  let domain: number[] // lower and upper bounds
-  let values: readonly [number, number?] // initial handle values
+  let domain: readonly [number, number] // lower and upper bounds
```
Comment (jpveooys) [inline packages/react-component-library/src/components/RangeSliderE/RangeSliderE.test.tsx:26]: `readonly [number, number?]` allowed `[number, undefined]`, which was incompatible with `readonly number[]`.

File: packages/react-component-library/src/components/RangeSliderE/SliderE.tsx [hunk 2]
```diff
@@ -42,13 +42,13 @@ export interface RangeSliderEProps extends Omit<SliderProps, SliderOmitType> {
    * Two element array of numbers providing the min and max values for the slider [min, max] e.g. [0, 100].
    * It does not matter if the slider is reversed on the screen, domain is always [min, max] with min < max.
    */
-  domain?: ReadonlyArray<number>
+  domain: readonly [number, number]
```
Comment (jpveooys) [inline packages/react-component-library/src/components/RangeSliderE/SliderE.tsx:45]: The component breaks without `domain` so I made this required.

File: packages/react-component-library/src/components/TopLevelNavigation/Masthead/useMastheadSearch.ts [hunk 3]
```diff
@@ -8,13 +8,17 @@ export function useMastheadSearch() {
...
+
     event.currentTarget.blur()
     const newShowSearch = !showSearch
 
     // if opening the searchbar then get the container width and set that
     // as the width of the searchbar so that it does not spill
     // over to other parts of the page, such as the sidebar.
-    if (newShowSearch === true) {
+    if (newShowSearch) {
       setContainerWidth(mastheadContainerRef.current.offsetWidth)
```
Comment (jpveooys) [inline packages/react-component-library/src/components/TopLevelNavigation/Masthead/useMastheadSearch.ts:22]: I noticed this doesn't react to container/viewport width changes – but out of scope for this issue.

File: packages/react-component-library/src/components/RangeSliderE/RangeSliderE.test.tsx [hunk 4]
```diff
@@ -23,8 +23,8 @@ describe('RangeSliderE', () => {
...
+  let values: readonly [number] | readonly [number, number] // initial handle values
```
Comment (jpveooys) [inline packages/react-component-library/src/components/RangeSliderE/RangeSliderE.test.tsx:27]: `readonly [number, number?]` allowed `[number, undefined]`, which was incompatible with `readonly number[]`.

Comment (m7kvqbe1) [reply packages/react-component-library/src/components/RangeSliderE/RangeSliderE.test.tsx:27]: tupletastic

**Review by m7kvqbe1 — APPROVED** [review]

**Review by thyhjwb6 — APPROVED** [review]
👍

**Pull request merged**